torch/csrc/lazy/core/shape_inference.cpp

```diff
@@ -418,6 +420,52 @@ std::vector<Shape> compute_shape_embedding_dense_backward(
       Shape(grad_output.scalar_type(), {num_weights, grad_output.size(-1)})};
 }
 
+std::vector<Shape> compute_shape_expand(
+    const at::Tensor& self,
+    at::IntArrayRef size,
+    bool implicit) {
+  CHECK_GE(size.size(), self.dim());
+  int64_t num_new_dimensions = size.size() - self.dim();
+  std::vector<int64_t> padded_self(num_new_dimensions, 0);
+  padded_self.insert(
+      padded_self.end(), self.sizes().begin(), self.sizes().end());
+  std::vector<int64_t> target_size(size.size());
+  for (const auto idx : c10::irange(size.size())) {
+    target_size[idx] = size[idx] == -1 ? padded_self[idx] : size[idx];
+  }
+  return {Shape(self.scalar_type(), target_size)};
+}
+
+std::vector<Shape> compute_shape_expand(
+    const at::Tensor& self,
+    c10::SymIntArrayRef size,
+    bool implicit) {
+  CHECK_GE(size.size(), self.dim());
+  std::vector<c10::SymInt> _sizes = ToVector<c10::SymInt>(size);
+  int64_t num_new_dimensions = _sizes.size() - self.dim();
+  std::vector<int64_t> padded_self(num_new_dimensions, 0);
+  padded_self.insert(
+      padded_self.end(), self.sizes().begin(), self.sizes().end());
+  std::vector<int64_t> target_size(_sizes.size());
+  for (const auto idx : c10::irange(_sizes.size())) {
+    if (_sizes[idx].is_symbolic()) {
+      std::shared_ptr<c10::SymbolicIntNode> symbolicIntNode =
+          _sizes[idx].toSymbolicIntNode();
+      auto lazySymIntNode =
+          std::dynamic_pointer_cast<torch::lazy::SymbolicIntNode>(
+              symbolicIntNode);
+      auto size_node = lazySymIntNode->node_;
+      auto static_value =
+          std::dynamic_pointer_cast<torch::lazy::DimensionNode>(size_node)
+              ->getStaticValue();
+      target_size[idx] = static_value;
+    } else {
+      target_size[idx] = _sizes[idx].data();
```

Comment:
I think we need an extra `if` in here: 

```cpp
} else {
  if (_sizes[idx].data() == -1) {
      // -1 can't be specified for non-existing dimensions
      TORCH_CHECK(idx >= num_new_dimensions);
       target_size[idx] =  padded_self[idx];
  } else {
     target_size[idx] =  _sizes[idx];
  }
}
```

Reply:
That's correct. 

I think we need to call the `.data()` method when assigning to `target_size[idx]`. Let me know if you think differently @Krovatkin.